stdlib/public/core/Collection.swift

```diff
@@ -1199,18 +1199,18 @@ extension Collection {
       return []
     }
 
-    var result = ContiguousArray<T>()
-    result.reserveCapacity(n)
-
-    var i = self.startIndex
-
-    for _ in 0..<n {
-      result.append(try transform(self[i]))
-      formIndex(after: &i)
+    return try unsafe Array<T>(unsafeUninitializedCapacity: n) { (
```

Comment:
@glessard I assume we'll update this to use an OutputSpan init once that's feasible?

Reply:
That would be preferable indeed.